Tune event listeners

freejoe76 opened the pull request:
Adjust event listeners so the ones assigned to the document only fire after the scrubber has started. Before the scrub change & end would fire on any mouse click on the document (see the console on http://interactive.nydailynews.com/map/school-shootings/before/ ), with this change the scrub change & end only fires on mouse clicks *after* the scrubber has been started (see the console on http://interactive.nydailynews.com/map/school-shootings/ ).

freejoe76 committed:
Tune event listeners

Adjust event listeners so the ones assigned to the document only fire after the scrubber has started. Before the scrub change & end would fire on any mouse click on the document (see the console on http://interactive.nydailynews.com/map/school-shootings/before/ ), with this change the scrub change & end only fires on mouse clicks *after* the scrubber has been started (see the console on http://interactive.nydailynews.com/map/school-shootings/ ).

freejoe76 committed:
Merge pull request #1 from freejoe76/freejoe76-patch-1

Tune event listeners

jwmerrill commented on scrubber.js:
```diff
@@ -130,6 +133,8 @@ ScrubberView.prototype.attachListeners = function ()  {
 
   var stop = function () {
+    document.removeEventListener('mouseup', stop, true);
+    document.removeEventListener('touchend', stop, true);
```
If I understand correctly, using the third "useCapture" parameter here is not desirable. Ref:

https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener

> Specifies whether the EventListener to be removed is registered as a capturing listener or not. If this parameter is absent, a default value of false is assumed.
>
> If a listener is registered twice, one with capture and one without, remove each one separately. Removal of a capturing listener does not affect a non-capturing version of the same listener, and vice versa.

So I think we'd only want to set the "useCapture" parameter to `true` here if we had also set it to `true` when we registered the event listener.

freejoe76 replied:
Ah, that makes sense. I was on that page earlier today, but missed that. I'll change and resubmit. Thanks!

jwmerrill commented:
Otherwise this looks great to me. Good find, and thanks a lot for the PR.

freejoe76 committed:
Unnecessary parameter

Per useCapture documentation on https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener

freejoe76 commented:
I updated the extra removeEventListener calls, per https://github.com/desmosinc/scrubber/pull/17/files/611e74675f450d50733eded15fed71a0f4c8642a#r169786459

jwmerrill commented:
Merged. Thanks again for this contribution.

freejoe76 commented:
Thanks for the code! Also, I just noticed the "Allow edits from maintainers." checkbox, which was unchecked, ha.
